metadata_backend/api/auth.py

```diff
@@ -136,8 +136,28 @@ async def callback(self, req: Request) -> Response:
                 reason="Could not set user, missing claim CSCUserName, remoteUserIdentifier or sub."
             )
 
+        # Handle projects, they come in different formats depending on the service used.
+        # Current project sources: CSC projects, LS AAI groups
+        projects: List[str] = []
+        if "sdSubmitProjects" in session["userinfo"]:
+            # CSC projects come in format "project1 project2"
+            projects = session["userinfo"]["sdSubmitProjects"].split(" ")
+        elif "eduperson_entitlement" in session["userinfo"]:
+            # LS AAI groups come in format ["group1", "group2"]
+            groups = session["userinfo"]["eduperson_entitlement"]
+            for group in groups:
+                # LSAAI groups are of format namespaces:root_group:actual_group#client
+                # and we want to get "root_group:actual_group", and any subgroups of actual_group
+                matched = re.search(f"(?<=)({self.lsaai_root}.*?)(?=#)", group)
+                if matched:
+                    projects.append(matched.group(0))
+        else:
+            # No project group information received, abort, as metadata-submitter
+            # object hierarchy depends on a project group to act as owner for objects
+            raise web.HTTPUnauthorized(reason="Insufficient user profile.")
```

Comment:
Perhaps the message could say that the user doesn't belong to any projects that have access to SD-Submit?

Reply:
Yeah, I was being very vague there, I'm sure we can give the actual reason